econml/tests/test_drlearner.py

```diff
@@ -0,0 +1,558 @@
+                                               ((d_y,) if d_y > 0 else ()) +
+                                               ((d_t_final,) if d_t_final > 0 else()))
+
+                # TODO: add stratification to bootstrap so that we can use it even with discrete treatments
+                infs = [None, 'statsmodels']
+
+                est = LinearDRLearner(model_regression=Lasso(),
+                                      model_propensity=LogisticRegression(C=1000, solver='lbfgs', multi_class='auto'))
+
+                for inf in infs:
+                    with self.subTest(d_w=d_w, d_x=d_x, d_y=d_y, d_t=d_t,
+                                      is_discrete=is_discrete, est=est, inf=inf):
+                        est.fit(Y, T, X, W, inference=inf)
+                        # make sure we can call the marginal_effect and effect methods
+                        const_marg_eff = est.const_marginal_effect(X)
+                        marg_eff = est.marginal_effect(T, X)
+                        self.assertEqual(shape(marg_eff), marginal_effect_shape)
+                        self.assertEqual(shape(const_marg_eff), const_marginal_effect_shape)
+
+                        np.testing.assert_array_equal(
+                            marg_eff if d_x else marg_eff[0:1], const_marg_eff)
+
+                        T0 = np.full_like(T, 'a')
+                        eff = est.effect(X, T0=T0, T1=T)
+                        self.assertEqual(shape(eff), effect_shape)
+                        if inf is not None:
+                            const_marg_eff_int = est.const_marginal_effect_interval(X)
+                            marg_eff_int = est.marginal_effect_interval(T, X)
+                            self.assertEqual(shape(marg_eff_int),
+                                             (2,) + marginal_effect_shape)
+                            self.assertEqual(shape(const_marg_eff_int),
+                                             (2,) + const_marginal_effect_shape)
+                            self.assertEqual(shape(est.effect_interval(X, T0=T0, T1=T)),
+                                             (2,) + effect_shape)
+
+                        est.score(Y, T, X, W)
+
+                        # make sure we can call effect with implied scalar treatments, no matter the
+                        # dimensions of T, and also that we warn when there are multiple treatments
+                        if d_t > 1:
+                            cm = self.assertWarns(Warning)
+                        else:
+                            cm = ExitStack()  # ExitStack can be used as a "do nothing" ContextManager
+                        with cm:
+                            effect_shape2 = (n if d_x else 1,) + ((d_y,) if d_y > 0 else())
+                            eff = est.effect(X, T0='a', T1='b')
+                            self.assertEqual(shape(eff), effect_shape2)
+
+    def test_can_use_vectors(self):

```

Comment:
Can you add a comment to each test which is virtually identical to a DML test so that we can consolidate them into generic tests in a later pass?